integrationtests/tests/fixtures/mod.rs

```diff
@@ -1025,6 +1044,23 @@ impl FederationTest {
         }
     }
 
+    pub async fn assert_peers_synchronized(&self) {
+        let full_server_status = self.full_server_status().await;
+        println!("assert_peers_synchronized: full_server_status: {full_server_status:?}");
+        assert!(full_server_status
+            .into_iter()
+            .all(|s| s.unwrap().peers_not_synchronized == 0));
+    }
+
+    pub async fn assert_peers_not_synchronized(&self) {
+        let full_server_status = self.full_server_status().await;
```

Comment:
Given the caching, lags etc. I don't think we should use status endpoints for asserting anything in tests.

Reply:
There is no caching (this is the method that the endpoints calls, not the result of the endpoint) and the information returned is the most up to date one regarding the internal state of the guardians.

But I agree that perhaps we shouldn't call it on tests right know because there is some point of synchronization missing (e.g maybe we still need to wait further things to happen until we are sure all the nodes are perfectly connected and have caught up the consensus)

Reply:
Oh, right. No caching, but indeed some synchronization might be needed. And also - this is supposed to be a diagnostic functionality. By utilizing it in tests, we now couple it with tests and other code, making potential changes more difficult and have a chance of breaking something more important that just status endpoint.

Reply:
Generally I agree. But then we are left with the question on how to really test this endpoint. It would be good to be sure it can detect issues on different scenarios. Adding it to existing integration tests seemed the most effective way.

But we can say that, for this particular case, unit tests and manual testing is enough.